examples/chef/chef.py

```diff
@@ -573,13 +583,18 @@ def main(argv: Sequence[str]) -> None:
         elif options.build_target == "linux":
             shell.run_cmd(f"cd {_CHEF_SCRIPT_PATH}/linux")
             with open(f"{_CHEF_SCRIPT_PATH}/linux/args.gni", "w") as f:
+                target_defines = [
+                    f"\"CHIP_DEVICE_CONFIG_DEVICE_VENDOR_ID={options.vid}\"", f"\"CHIP_DEVICE_CONFIG_DEVICE_PRODUCT_ID={options.pid}\""]
+                target_defines.append(f"\"CONFIG_ENABLE_PW_RPC={'1' if options.do_rpc else '0'}\"")
+                if options.do_automated_test_stamp:
+                    target_defines.append(f"\"CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING=\\\"{sw_ver_string}\\\"\"")
                 f.write(textwrap.dedent(f"""\
                         import("//build_overrides/chip.gni")
                         import("${{chip_root}}/config/standalone/args.gni")
                         chip_shell_cmd_server = false
                         chip_build_libshell = true
                         chip_config_network_layer_ble = false
-                        target_defines = ["CHIP_DEVICE_CONFIG_DEVICE_VENDOR_ID={options.vid}", "CHIP_DEVICE_CONFIG_DEVICE_PRODUCT_ID={options.pid}", "CONFIG_ENABLE_PW_RPC={'1' if options.do_rpc else '0'}", "CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING=\\"{branch}:{commit_id}\\""]
+                        target_defines = [{", ".join(target_defines)}]
```

Comment:
Please declare these as GN arguments via declare_args() and set them that way.

Do not use target_defines to directly set options.

Reply:
The target_defines here involving more things than what we want to deliver in this change(CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING).

How about to let me make the fixes in the another changes?

Reply:
As discussed, I'll create a new GN argument for this new config, and leave original target_defines as is.
Thank you!